src/web/components/SignInGate/displayRule.ts

```diff
@@ -79,15 +83,53 @@ export const canShow = (
 	CAPI: CAPIBrowserType,
 	isSignedIn: boolean,
 	currentTest: CurrentABTest,
-): boolean =>
-	!isSignedIn &&
-	!hasUserDismissedGateMoreThanCount(
-		currentTest.variant,
-		currentTest.name,
-		5,
-	) &&
-	isNPageOrHigherPageView(3) &&
-	isValidContentType(CAPI) &&
-	isValidSection(CAPI) &&
-	isValidTag(CAPI) &&
-	!isIOS9();
+): Promise<boolean> => {
+	return Promise.resolve(
+		!isSignedIn &&
+			!hasUserDismissedGateMoreThanCount(
+				currentTest.variant,
+				currentTest.name,
+				5,
+			) &&
+			isNPageOrHigherPageView(3) &&
+			isValidContentType(CAPI) &&
+			isValidSection(CAPI) &&
+			isValidTag(CAPI) &&
+			!isIOS9(),
+	);
```

Comment:
I'm using `Promise.resolve` to lift a boolean to a promise. I'm not sure how you're thinking of using `Promise.all` here

Reply:
I assume the use of a `Promise` was because some of these functions are async? 
if so, you would benefit by using `Promise.all` to call them all at once rather than individually?
If none of them are async do you need to return a `Promise`?

Reply:
`SignInGateComponent` requires a canShow function that returns a Promise. Some implementations of canShow are async. I have lifted the implementations that are not.